### DISCUSSION TITLE
How to: Show an Image on ButtonWidget.

### DISCUSSION THREAD

**Discussion by uSafi** [Mapsui/Mapsui#2646] [Q&A] [answered]
Hello
How to get an embedded resource Image like png or svg into ButtonWidget?
or even an Image with it's Build Action set to "MauiImage" or "Content .. Copy Always"
I know about using those lines:

            buttonWidget.SvgImage = imagePath;
            buttonWidget.Picture ??= buttonWidget.SvgImage?.LoadSvgPicture();  

But no matter how i get the path
examples:
"Resources.Images.test.svg"
"Resources/Images/test.svg"
"Resources\Images\test.svg"
$"{assemblyName}.Resources.Images.test.png"
$"{assemblyName}.Resources.Images.test.svg"
and more tries...

once this line run 

        buttonWidget.Picture ??= buttonWidget.SvgImage?.LoadSvgPicture();
        
I always end up with this error:
"Data at the root level is invalid. Line 1, position 1."

I appreciate any help.

**Comment by uSafi** [comment, marked as answer]
I found the way:
    
    buttonWidget.Text = " ";
    buttonWidget.Picture = EmbeddedResourceLoader.Load("Resources.Images.refresh.svg", typeof(App)).LoadSvgPicture();
    
Don't ask me why the use of buttonWidget.Text = " ";
For now, the image doesn't show without putting something in the Text Property.

**Reply by pauldendulk** [reply]
The `buttonWidget.Text = " "` thing is a bug that is solved in the upcoming v5. 

Loading the image will become this: `buttonWidget.ImageSource = "embedded://MyAppProjectName.Resources.Images.refresh.svg"`.

The way users are supposed to find the answer to your question is through the samples. In this case the ButtonSample. We need to make it easier for users to find this.

**Reply by uSafi** [reply]
> The `buttonWidget.Text = " "` thing is a bug that is solved in the upcoming v5.
> 
> Loading the image will become this: `buttonWidget.ImageSource = "embedded://MyAppProjectName.Resources.Images.refresh.svg"`.
> 
> The way users are supposed to find the answer to your question is through the samples. In this case the ButtonSample. We need to make it easier for users to find this.

Thank you, Sir
The upcoming v5 looks interesting.

Best Regards
